src/components/Statistics.tsx

```diff
@@ -38,53 +38,83 @@ interface Props {
 const Statistics = ({ refresh }: Props) => {
   const { client } = useGlobalContext()
   const [hashrate, setHashrate] = useState('')
+  const [isHashrateLoading, setIsHashrateLoading] = useState(true)
   const [totalSupply, setTotalSupply] = useState<number>()
+  const [isTotalSupplyLoading, setIsTotalSupplyLoading] = useState(true)
   const [circulating, setCirculating] = useState<number>()
+  const [isCirculatingLoading, setIsCirculatingLoading] = useState(true)
   const [transactions, setTransactions] = useState<number>()
+  const [isTransactionsLoading, setIsTransactionsLoading] = useState(true)
   const [blocks, setBlocks] = useState<number>()
+  const [isBlocksLoading, setIsBlocksLoading] = useState(true)
   const [avgBlockTime, setAvgBlockTime] = useState<number>()
-  const [isLoading, setIsLoading] = useState(true)
+  const [isAvgBlockTimeLoading, setIsAvgBlockTimeLoading] = useState(true)
 
   const currentSupplyPercentage = circulating && totalSupply && ((circulating / totalSupply) * 100).toPrecision(3)
 
   useEffect(() => {
     if (!client) return
 
-    const fetchData = async () => {
+    const fetchHashrateData = async () => {
       const now = new Date().getTime()
       const yesterday = now - ONE_DAY
 
       await client.charts
         .getChartsHashrates({ fromTs: yesterday, toTs: now, 'interval-type': 'hourly' })
         .then(({ data }) => setHashrate(data.length > 0 ? data[0].value : ''))
 
+      setIsHashrateLoading(false)
+    }
+
+    const fetchBlocksData = async () => {
       await client.infos
         .getInfosHeights()
         .then(({ data }) => setBlocks(data.reduce((acc: number, { value }) => acc + value, 0)))
 
+      setIsBlocksLoading(false)
+    }
+
+    const fetchTotalSupplyData = async () => {
```

Comment:
Following the comment above, the next 3 fn could be a bit dried up 🌞 
Again, some pseudo code:

```ts
fetchData = async (call: () => Promise<HttpResponse>, statKey: StatKey) => {
  await call()
  .then((res) => res.text())
  .then((text) => updateDataState({[statKey]: parseInt(text)})
}
```

Reply:
There 😛